### ISSUE TITLE
Add pagination support to file edit view

### ISSUE THREAD

**Issue by poikilotherm** [IQSS/dataverse#7371]
In #4656 / #4881, pagination for the file table has been added.

Now I noticed when uploading a ZIP file with 3,000 files (enabled that via setting) that the browser tab crashes because the file table in edit view is getting way to big.

Can we please add pagination support for this, too? As always, happy to contribute.

Pinging @TaniaSchlatter as this is UI related, @mheppler and @pdurbin because they wrote the pagination bits for the dataset view.

**Comment by scolapasta** [comment]
Could you clarify a little more of what you're trying to edit. Currently, the way edit works is that you have to check the files you want to edit, so you don't have to edit all the files at once. But I'm assuming you do in this case.

We do allow via the checkboxes, for example, the ability to bulk restrict / unrestrict, delete, etc. Is there something in particular on the edit page? if for example, you wanted to edit all the paths, pagination wouldn'y be the idea solution, because you would have to manually type them all in. So we'd be better off allowing some sort of bulk path edit.

Another challenge with pagination for edit is validation. For example, edit something on page 1, go to to page 2, edit something, click save. How would we handle validation issues for changes on page 1 (or for both 1 and 2)?

**Comment by poikilotherm** [comment]
- Configured Dataverse to accept ZIP files containing 3000 files.
- Created an empty dataset.
- Went to "Upload files", choosed the ZIP and watched the upload.
- Upload, extraction and thumbnail generation took about 25 secs (that's fine) on a local disk. (NFS ~4 minutes)

![grafik](https://user-images.githubusercontent.com/5468333/97599175-fb692f80-1a07-11eb-9e21-8cad36cf7b07.png)
![grafik](https://user-images.githubusercontent.com/5468333/97599217-0328d400-1a08-11eb-993b-dec69d69857b.png)
![grafik](https://user-images.githubusercontent.com/5468333/97599308-20f63900-1a08-11eb-9403-661bd7250629.png)

Finally, after 1 CPU core for ~5 minutes + 2 GiB RAM for this Firefox tab, I get this (only a section at the bottom of course):

![grafik](https://user-images.githubusercontent.com/5468333/97599984-de812c00-1a08-11eb-816d-34fa1a986ccc.png)

- Unsorted entries (watch the numbers, they should be `rnd_(1-3000).png`)
- I had a browser tab crash once. Maybe too much memory.

**Comment by scolapasta** [comment]
AH, I see, you mean the edit at the time of upload. I was thinking you had uploaded and wanted to do edits after (which do use the same page).

**Comment by poikilotherm** [comment]
FWIW I created a runtime analysis with Firefox and saved the profile. Be warned, it's big, but you should be able to load it inside the dev tools to review.

![grafik](https://user-images.githubusercontent.com/5468333/97601795-d9bd7780-1a0a-11eb-8e96-aea74a2d7ac1.png)

The browser is pretty active with recalculating style...

[profile.zip](https://github.com/IQSS/dataverse/files/5459796/profile.zip)

**Comment by cmbz** [comment]
2024/11/04: Adding to SPA future improvements list.
